### PR TITLE
Update to cuda 11.8

### PR DESCRIPTION
**Pull request by jchen351**
### Description
Fixed [AB#14497](https://aiinfra.visualstudio.com/6a833879-cd9b-44a4-a9de-adc2d818f13c/_workitems/edit/14497)

### CODE REVIEW

**Comment by snnn** [comment]
@jchen351 , how is it going? Did you try to run the nightly packaging pipeline?

**Comment by jchen351** [comment]
> @jchen351 , how is it going? Did you try to run the nightly packaging pipeline?

Just started.

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 1]
```diff
@@ -126,7 +126,7 @@ stages:
     buildArch: x64
     msbuildPlatform: x64
     packageName: x64-tensorrt
-    buildparameter: --use_tensorrt --use_tensorrt_builtin_parser --tensorrt_home="C:\local\TensorRT-8.6.0.12.Windows10.x86_64.cuda-11.8" --cuda_version=11.8 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8"  --enable_onnx_tests --enable_wcos --build_java --cmake_extra_defines "CMAKE_CUDA_ARCHITECTURES=52;60;61;70;75;80"
+    buildparameter: --use_tensorrt --use_tensorrt_builtin_parser --tensorrt_home="C:\local\TensorRT-8.6.0.12.Windows10.x86_64.cuda-11.8" --cuda_version=11.6 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6"  --enable_onnx_tests --enable_wcos --build_java --cmake_extra_defines "CMAKE_CUDA_ARCHITECTURES=52;60;61;70;75;80"
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml]: why revert back?

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml]: I asked him to limit this change to Linux only.

**Review by snnn — APPROVED** [review]

**Pull request merged**